### PR TITLE
delete unused functions in pkg/apis directory

### PR DESCRIPTION
**Pull request by HirazawaUi**
#### What type of PR is this?

/kind cleanup

#### Which issue(s) this PR fixes:
#115088 
#111666

#### Special notes for your reviewer:
This PR follows the content of #111654, the purpose is to split #111654 into many small PRs, this PR only modifies the pkg/apis directory

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by HirazawaUi** [comment]
@thockin please review this PR, does it still too big?

File: pkg/apis/admission/register.go [hunk 1]
```diff
@@ -27,16 +27,6 @@ const GroupName = "admission.k8s.io"
 // SchemeGroupVersion is group version used to register these objects
 var SchemeGroupVersion = schema.GroupVersion{Group: GroupName, Version: runtime.APIVersionInternal}
 
-// Kind takes an unqualified kind and returns a Group qualified GroupKind
-func Kind(kind string) schema.GroupKind {
```
Comment (thockin) [inline pkg/apis/admission/register.go:31]: These functions represent a sort of API to the api packages.  It sure seems like this is not used, but I don't know why it was there.

@lavalamp or @deads2k or @jpbetz - can we get a consult?

**Comment by HirazawaUi** [comment]
the two e2e test failures don't seem to be related to this PR, can we start test again?

File: pkg/apis/admission/register.go (hunk 1 above)
Comment (lavalamp) [reply pkg/apis/admission/register.go:31]: I would not remove these unless you can show that no one uses the corresponding function from any api package.

**Comment by lavalamp** [comment]
Please don't remove the api functions in the same PR as the other functions, that makes it super hard to review.

**Comment by HirazawaUi** [comment]
> Please don't remove the api functions in the same PR as the other functions, that makes it super hard to review.

ok, I will split the api functions and other functions in different PRs

**Comment by HirazawaUi** [comment]
@lavalamp Hi, this PR now only keeps remove api functions

**Comment by lavalamp** [comment]
Thanks for splitting.

I currently believe we should keep these functions, to keep the packages consistent, even if not every package's version of the function happens to get called.

**Comment by thockin** [comment]
There's not an easy way to express "this is part of a package interface", but I can buy what @lavalamp is saying.

OTOH - these all seem to be implemented identically in terms of the already-public `<pkg>.SchemeGroupVersion` - could we replace all of them with a single function in a common place, like

```
func Kind(gv schema.GroupVersion) schema.GroupKind {		
    return gv.WithKind(kind).GroupKind()		
}
```

?

**Comment by lavalamp** [comment]
Removing the functions everywhere even if they are used and replacing it with something is fine with me, but seems like a lot of work.

**Comment by thockin** [comment]
I agree, just making sure we're on the same page.

File: pkg/apis/admission/register.go (hunk 1 above)
Comment (thockin) [reply pkg/apis/admission/register.go:31]: As discussed in comments - we should leave these ones.  They represent a package-api.

**Pull request closed without merging**